### PR TITLE
feat(vertexai): Add support for Storage prompt and responseMimeType

### PR DESCRIPTION
**Pull request by cynthiajoan**
## Description

- bump google_generative_ai to 0.4.0
- Add Firebase Storage url support
- Add responseMimeType

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by a-maurice — APPROVED** [review]

File: packages/firebase_vertexai/firebase_vertexai/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 ## 0.0.1
```
Comment (natebosch) [inline packages/firebase_vertexai/firebase_vertexai/CHANGELOG.md:1]: I'd go with `0.1.0`. Using a leading `0.0` has caused some confusion in the past. There are no downsides to starting at `0.1.0`.

File: packages/firebase_vertexai/firebase_vertexai/lib/firebase_vertexai.dart [hunk 2]
```diff
@@ -25,6 +25,8 @@ import 'package:firebase_core_platform_interface/firebase_core_platform_interfac
 import 'package:google_generative_ai/google_generative_ai.dart' as google_ai;
 // ignore: implementation_imports, tightly coupled packages
 import 'package:google_generative_ai/src/model.dart' as google_ai_model;
+// ignore: implementation_imports, tightly coupled packages
+import 'package:google_generative_ai/src/vertex_hooks.dart';
```
Comment (natebosch) [inline packages/firebase_vertexai/firebase_vertexai/lib/firebase_vertexai.dart:29]: You should be able to also remove the above import and use the `createModel` API from this library.

File: packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_api.dart [hunk 3]
```diff
@@ -44,6 +37,13 @@ final class CountTokensResponse {
   final int? totalBillableCharacters;
 }
 
+/// Extension on [google_ai.CountTokensResponse] to access extra fields
+extension CountTokensResponseFields on google_ai.CountTokensResponse {
+  /// total billable Characters for the prompt.
```
Comment (natebosch) [inline packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_api.dart:42]: [nit] 
```suggestion
  /// Total billable Characters for the prompt.
```

**Review by natebosch — APPROVED** [review]

File: packages/firebase_vertexai/firebase_vertexai/lib/firebase_vertexai.dart (hunk 2 above)
Comment (cynthiajoan) [reply packages/firebase_vertexai/firebase_vertexai/lib/firebase_vertexai.dart:29]: done

File: packages/firebase_vertexai/firebase_vertexai/CHANGELOG.md (hunk 1 above)
Comment (cynthiajoan) [reply packages/firebase_vertexai/firebase_vertexai/CHANGELOG.md:1]: done

**Pull request merged**